src/plist.c

```diff
@@ -67,12 +67,18 @@ Int             GrowPlist (
     UInt                plen;           /* new physical length             */
     UInt                good;           /* good new physical length        */
 
+    if (!UINT_FITS_IN_INTOBJ(need)) {
+        ErrorMayQuit("GrowPlist: List size too large", 0, 0);
+    }
+
     /* find out how large the plain list should become                     */
     good = 5 * (SIZE_OBJ(list)/sizeof(Obj)-1) / 4 + 4;
 
     /* but maybe we need more                                              */
-    if ( need < good ) { plen = good; }
-    else               { plen = need; }
+    if (need < good && UINT_FITS_IN_INTOBJ(good))
```

Comment:
Then this can be simplified to 
```
    if (need < good)
```
In your version adding and appending to lists of length > 4/5 * maximal length becomes very inefficient.

Reply:
Excellent point, thanks, I'll update this PR.